apps/dav/lib/Service/AbsenceService.php

```diff
@@ -81,27 +81,32 @@ public function createOrUpdateAbsence(
 		}
 
 		$now = $this->timeFactory->getTime();
-		if ($eventData->getStartDate() > $now) {
-			$this->jobList->scheduleAfter(
-				OutOfOfficeEventDispatcherJob::class,
-				$eventData->getStartDate(),
-				[
-					'id' => $absence->getId(),
-					'event' => OutOfOfficeEventDispatcherJob::EVENT_START,
-				],
-			);
-		}
-		if ($eventData->getEndDate() > $now) {
-			$this->jobList->scheduleAfter(
-				OutOfOfficeEventDispatcherJob::class,
-				$eventData->getEndDate(),
-				[
-					'id' => $absence->getId(),
-					'event' => OutOfOfficeEventDispatcherJob::EVENT_END,
-				],
-			);
+		if($eventData->getEndDate() < $now) {
+			// absence is in the past
+			return $absence;
 		}
 
+		// If the absence is for today the timestamp will be smaller than $now
+		// so run the job now if that is the case
+		$runJobAt = ($eventData->getStartDate() < $now) ? $now : $eventData->getStartDate();
+		$this->jobList->scheduleAfter(
+			OutOfOfficeEventDispatcherJob::class,
+			$runJobAt,
+			[
+				'id' => $absence->getId(),
+				'event' => OutOfOfficeEventDispatcherJob::EVENT_START,
+			],
+		);
```

Comment:
```suggestion
		$this->jobList->scheduleAfter(
			OutOfOfficeEventDispatcherJob::class,
			$eventData->getStartDate(),
			[
				'id' => $absence->getId(),
				'event' => OutOfOfficeEventDispatcherJob::EVENT_START,
			],
		);
```

Can always use `getStartDate()` here as the job will be picked up if it is in the past IIRC.